src/ints/mouse.cpp

```diff
@@ -509,76 +559,22 @@ void Mouse_CursorMoved(float xrel,float yrel,float x,float y,bool emulate) {
 		if (mouse.y >= 32768.0) mouse.y -= 65536.0;
 		else if (mouse.y <= -32769.0) mouse.y += 65536.0;
 	}
-	Mouse_AddEvent(MOUSE_HAS_MOVED);
+	AddEvent(DOS_EV::MOUSE_HAS_MOVED);
 	DrawCursor();
 }
 
-void Mouse_CursorSet(float x,float y) {
-	mouse.x=x;
-	mouse.y=y;
-	DrawCursor();
-}
-
-void Mouse_ButtonPressed(uint8_t button) {
-	switch (button) {
-#if (MOUSE_BUTTONS >= 1)
-	case 0:
-		if (mouse.buttons&1) return;
-		mouse.buttons|=1;
-		Mouse_AddEvent(MOUSE_LEFT_PRESSED);
-		break;
-#endif
-#if (MOUSE_BUTTONS >= 2)
-	case 1:
-		if (mouse.buttons&2) return;
-		mouse.buttons|=2;
-		Mouse_AddEvent(MOUSE_RIGHT_PRESSED);
-		break;
-#endif
-#if (MOUSE_BUTTONS >= 3)
-	case 2:
-		if (mouse.buttons&4) return;
-		mouse.buttons|=4;
-		Mouse_AddEvent(MOUSE_MIDDLE_PRESSED);
-		break;
-#endif
-	default:
-		return;
-	}
-	mouse.times_pressed[button]++;
-	mouse.last_pressed_x[button]=POS_X;
-	mouse.last_pressed_y[button]=POS_Y;
+static inline uint8_t GetResetWheel8bit() {
+    if (!mouse.cute_mouse) return 0;
+    int8_t tmp = static_cast<int8_t>(std::clamp(mouse.wheel, static_cast<int16_t>(-0x80), static_cast<int16_t>(0x7f)));
```

Comment:
const

Reply:
This clamping logic seems familiar from somewhere else, could reuse the same function here if you extracted it.

Reply:
Added const. Unfortunately, the std::clamp usages are often not the same (they are related to int8_t, int16_t, or float; sometimes they are against INTxx_MAX, sometimes against other variables), moreover there are in different files. I was able to put them in a lambda (for example) in some places, but nothing really general is feasible.